packages/contracts/deploy/018-fund-accounts.ts

```diff
@@ -0,0 +1,66 @@
+/* Imports: External */
+import { sleep } from '@eth-optimism/core-utils'
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import {
+  defaultHardhatNetworkHdAccountsConfigParams,
+  defaultHardhatNetworkParams,
+} from 'hardhat/internal/core/config/default-config'
+import { normalizeHardhatNetworkAccountsConfig } from 'hardhat/internal/core/providers/util'
+
+/* Imports: Internal */
+import { getDeployedContract } from '../src/hardhat-deploy-ethers'
+
+// This is a TEMPORARY way to fund the default hardhat accounts on L2. The better way to do this is
+// to make a modification to hardhat-ovm. However, I don't have the time right now to figure the
+// details of how to make that work cleanly. This is fine in the meantime.
+const deployFn: DeployFunction = async (hre) => {
+  // Only execute this step if we're on the hardhat chain ID.
+  const { chainId } = await hre.ethers.provider.getNetwork()
+  if (chainId === defaultHardhatNetworkParams.chainId) {
+    const Proxy__OVM_L1ETHGateway = await getDeployedContract(
+      hre,
+      'Proxy__OVM_L1ETHGateway',
+      {
+        iface: 'OVM_L1ETHGateway',
+      }
+    )
+
+    const accounts = normalizeHardhatNetworkAccountsConfig(
+      defaultHardhatNetworkHdAccountsConfigParams
+    )
+
+    // Fund the accounts in parallel to speed things up.
```

Comment:
How many accounts is this? Is it the configured amount in the hardhat config? Just wondering if this could ever accidentally be set too high

Reply:
It's 20 accounts. I doubt it would be set much higher by hardhat.

Reply:
Ok if it runs without a problem then its not a problem. If its slow then perhaps we only do a subset of the 20

Reply:
I will restrict to 20 accounts with .slice